### PR TITLE
FIxing the settings_timeout_test flake by reducing poll deadline value

### PR DESCRIPTION
**Pull request by donnadionne**

### CODE REVIEW

**Comment by jtattermusch** [comment]
You don't need to submit this PR to be able to debug and see if changing the timeout helps:

Just running `bazel --bazelrc=tools/remote_build/windows.bazelrc test --config=windows_opt //test/core/transport/chttp2:settings_timeout_test --runs_per_test 100` from your windows machine should give you all the answers you need.

File: test/core/transport/chttp2/settings_timeout_test.cc [hunk 1]
```diff
@@ -111,7 +111,7 @@ class Client {
     EventState state;
     grpc_tcp_client_connect(state.closure(), &endpoint_, pollset_set,
                             nullptr /* channel_args */, server_addresses->addrs,
-                            1000);
+                            2000);
```
Comment (jtattermusch) [inline test/core/transport/chttp2/settings_timeout_test.cc]: based on my testing (running bazel RBE from my workstation), increasing the timeout actually fixes the problem (I got 500 successful runs without a failure), but I'd like to understand more why increasing this timeout actually helps (are we just increasing the timeout blindly or is there some reasoning behind this?)

https://source.cloud.google.com/results/invocations/6a11ddab-7617-47d6-b1b5-74a63d3e17b2/targets/%2F%2Ftest%2Fcore%2Ftransport%2Fchttp2:settings_timeout_test/tests

**Comment by jtattermusch** [comment]
Here is a log from 100 runs, out of which 13 have failed (with the current timeout of 1000ms).
https://source.cloud.google.com/results/invocations/f6059109-b772-48e4-af24-0eeae8712763/targets/%2F%2Ftest%2Fcore%2Ftransport%2Fchttp2:settings_timeout_test;shard=1;run=15/log

The issue I see is that in most the latency of connect is very low in case of success  and very close to 1sec if it fails.  That might indicate a problem with the way we kick on windows - it seems that despite the socket has already connected, we don't check for the status of the socked until 1s later (which is already too late because of the 1s connect timeout) and fail. This also explains why with 2sec timeout things work fine.

**Comment by jtattermusch** [comment]
> Here is a log from 100 runs, out of which 13 have failed (with the current timeout of 1000ms).
> https://source.cloud.google.com/results/invocations/f6059109-b772-48e4-af24-0eeae8712763/targets/%2F%2Ftest%2Fcore%2Ftransport%2Fchttp2:settings_timeout_test;shard=1;run=15/log
> 
> The issue I see is that in most the latency of connect is very low in case of success and very close to 1sec if it fails. That might indicate a problem with the way we kick on windows - it seems that despite the socket has already connected, we don't check for the status of the socked until 1s later (which is already too late because of the 1s connect timeout) and fail. This also explains why with 2sec timeout things work fine.

Indeed, when the test passes, on_connect is invoked almost immediately after starting the connection (usually <10ms later), but when the test fails, the on_connect is only called after one second. I think something is wrong with polling and we basically get the notification about successful connect too late.

**Comment by jtattermusch** [comment]
Please provide:
1. some reasoning behind this change (analysis of what goes wrong in the test when it flakes and how is the change supposed to fix the stuff - otherwise 1 year from now when someone looks at the PR, we will have no understanding why we changed what we changed).
2. did you confirm this fixes the flake? how many runs have you done?

**Comment by donnadionne** [comment]
I included the following in the commit comments; let me know if you have further questions.

Intermittently (when we run the test 100 times) we see that at times
we are not notified about successful socket connection even within 1
second.

The pollset_work deadline is set to 1 second, so when our connection
timeout is also set to 1 second, we only had 1 poll.  And if that poll
didn't let us know about the socket, we didn't get another chance.

Reducing the pollset_work deadline to 100ms, now the test runs faster
and pass consistently:

result for 500 runs:
https://source.cloud.google.com/results/invocations/dd1b38f4-0f28-470b-a78b-b485445e44f2/targets/%2F%2Ftest%2Fcore%2Ftransport%2Fchttp2:settings_timeout_test/tests

**Pull request merged**